File: src/main/java/org/example/lab1/entities/dto/PersonDTO.java
package org.example.lab1.entities.dto;

import org.example.lab1.entities.dao.Color;
import org.example.lab1.entities.dao.Country;
import org.example.lab1.entities.dao.Person;

public record PersonDTO(long id,
                        String name,
                        long coordinatesId,
                        java.util.Date creationDate,
                        Color eyeColor,
                        Color hairColor,
                        long locationId,
                        Float height,
                        java.time.LocalDateTime birthday,
                        Float weight,
                        Country nationality){
    public Person toDAO() {
        Person newPerson =  new Person();
        newPerson.setId(id);
        newPerson.setName(name);
        newPerson.setCreationDate(creationDate);
        newPerson.setEyeColor(eyeColor);
        newPerson.setHairColor(hairColor);
        newPerson.setHeight(height);
        newPerson.setWeight(weight);
        newPerson.setBirthday(birthday);
        newPerson.setNationality(nationality);
        return newPerson;
    }

}

File: frontend/src/dtos/PersonDTO.ts
import Color from "./ColorEnum";
import Country from "./CountryEnum";

interface PersonDTO {
    id?: number;
    name: string;
    coordinatesId: number;
    creationDate?: string;
    eyeColor?: Color;
    hairColor: Color;
    locationId?: number;
    height: number;
    birthday: string;
    weight?: number;
    nationality: Country;
}

export default PersonDTO;